tests/codex/sales/states/testinitialproving.nim

```diff
@@ -47,30 +57,57 @@ asyncchecksuite "sales state 'initialproving'":
     let next = state.onFailed(request)
     check !next of SaleFailed
 
-  test "switches to filling state when initial proving is complete":
-    let next = await state.run(agent)
-    check !next of SaleFilling
-    check SaleFilling(!next).proof == proof
+  test "waits for the beginning of the period to get the challenge":
+    let future = state.run(agent)
+    await sleepAsync(10.millis)
+    check not future.finished
+    await allowProofToStart()
+    discard await future
+
+  test "waits another period when the proof pointer is about to wrap around":
+    market.proofPointer = 250
+    let future = state.run(agent)
+    await allowProofToStart()
+    await sleepAsync(10.millis)
+    check not future.finished
+    market.proofPointer = 100
+    await allowProofToStart()
+    discard await future
```

Comment:
I feel like this needs a (~1ms) timeout, because this test is effectively testing that the state completes immediately given that the period has advanced.

Reply:
I don't understand? After advancing the period, we `await` the future. This is allowed to take a while (actually it is allowed to take forever, because we didn't specify a timeout).